tests/e2e/csi_snapshot_utils.go

```diff
@@ -0,0 +1,593 @@
+func createPreProvisionedSnapshotInGuestCluster(ctx context.Context, volumeSnapshot *snapV1.VolumeSnapshot,
+	updatedSnapshotContent *snapV1.VolumeSnapshotContent,
+	snapc *snapclient.Clientset, namespace string, pandoraSyncWaitTime int,
+	svcVolumeSnapshotName string) (*snapV1.VolumeSnapshotContent, *snapV1.VolumeSnapshot, error) {
+
+	framework.Logf("Change the deletion policy of VolumeSnapshotContent from Delete to Retain " +
+		"in Guest Cluster")
+	updatedSnapshotContent, err := changeDeletionPolicyOfVolumeSnapshotContentOnGuest(ctx, updatedSnapshotContent,
+		snapc, namespace)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to change deletion policy of VolumeSnapshotContent: %v", err)
+	}
+
+	framework.Logf("Delete dynamic volume snapshot from Guest Cluster")
+	deleteVolumeSnapshotWithPandoraWait(ctx, snapc, namespace, volumeSnapshot.Name, pandoraSyncWaitTime)
+
+	framework.Logf("Delete VolumeSnapshotContent from Guest Cluster explicitly")
+	err = deleteVolumeSnapshotContentOnGuest(ctx, updatedSnapshotContent, snapc, namespace, pandoraSyncWaitTime)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to delete VolumeSnapshotContent: %v", err)
+	}
+
+	framework.Logf(fmt.Sprintf("Creating static VolumeSnapshotContent in Guest Cluster using "+
+		"supervisor VolumeSnapshotName %s", svcVolumeSnapshotName))
+	staticSnapshotContent, err := snapc.SnapshotV1().VolumeSnapshotContents().Create(ctx,
+		getVolumeSnapshotContentSpec(snapV1.DeletionPolicy("Delete"), svcVolumeSnapshotName,
+			"static-vs", namespace), metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create static VolumeSnapshotContent: %v", err)
+	}
+
+	framework.Logf("Verify VolumeSnapshotContent is created or not in Guest Cluster")
+	staticSnapshotContent, err = snapc.SnapshotV1().VolumeSnapshotContents().Get(ctx,
+		staticSnapshotContent.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get static VolumeSnapshotContent: %v", err)
+	}
+	framework.Logf("Snapshotcontent name is  %s", staticSnapshotContent.ObjectMeta.Name)
+
+	if !*staticSnapshotContent.Status.ReadyToUse {
+		return nil, nil, fmt.Errorf("VolumeSnapshotContent is not ready to use")
+	}
+
+	ginkgo.By("Create a static volume snapshot by static snapshotcontent")
+	staticVolumeSnapshot, err := snapc.SnapshotV1().VolumeSnapshots(namespace).Create(ctx,
+		getVolumeSnapshotSpecByName(namespace, "static-vs",
+			staticSnapshotContent.ObjectMeta.Name), metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create static volume snapshot: %v", err)
+	}
+	framework.Logf("Volume snapshot name is : %s", staticVolumeSnapshot.Name)
+
+	ginkgo.By("Verify static volume snapshot is created")
+	staticSnapshot, err := waitForVolumeSnapshotReadyToUse(*snapc, ctx, namespace, staticVolumeSnapshot.Name)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to wait for volume snapshot: %v", err)
```

Comment:
Can we fail with this error message: "volumeSnapshot is still not ready to use"

Reply:
done